src/ert/config/scalar_parameter.py

```diff
@@ -0,0 +1,515 @@
+            aggregate_function=None,
+        ).with_columns(pl.lit("transformed").alias("type"))
+        return df_raw.vstack(df_transformed)
+
+    def sample_or_load(
+        self,
+        real_nr: int,
+        random_seed: int,
+        ensemble_size: int,
+    ) -> pl.DataFrame:
+        df_sampled = self._sample_value(
+            self.scalar_params[DataSource.SAMPLED],
+            str(random_seed),
+            real_nr,
+        )
+        # load design matrix values and append them to the sampled values
+        # todo
+        return df_sampled
+
+    @staticmethod
+    def load_parameters(
+        ensemble: Ensemble, group: str, realizations: npt.NDArray[np.int_]
+    ) -> pl.DataFrame:
+        return pl.DataFrame(
+            [
+                ensemble.load_parameters_scalar(realization)
+                for realization in realizations
+            ]
+        )
+
+    def save_parameters(
+        self,
+        ensemble: Ensemble,
+        group: str,
+        realization: int,
+        data: npt.NDArray[np.float64],
+    ) -> None:
+        pass
+        # num_rows = 10000  # Number of rows
+        # group_names = ['GroupA', 'GroupB', 'GroupC']
+        # key_names = ['Key1', 'Key2', 'Key3']
+
+        # # Create data for each group-key combination
+        # data = {}
+        # for group in group_names:
+        #     for key in key_names:
+        #         col_name = f"{group}_{key}"  # Combine group and key names
+        #         data[col_name] = np.random.randint(0, 100, size=num_rows)
+
+        # # Create Polars DataFrame
+        # df = pl.DataFrame(data)
+        # ensemble.save_parameters(group, realization, ds)
+
+    def __len__(self) -> int:
+        return sum(len(params) for params in self.scalar_params.values())
+
+    def read_from_runpath(
+        self,
+        run_path: Path,
+        real_nr: int,
+        iteration: int,
+    ) -> None:
+        """
+        forward_init will not be supported, so None for the moment
+        """
+        return None
+
+    def write_to_runpath(
+        self, run_path: Path, real_nr: int, ensemble: Ensemble
+    ) -> dict[str, dict[str, float]] | None:
+        """
+        This function is responsible for converting the parameter
+        from the internal ert format to the format the forward model
+        expects
+        """
+        df = ensemble.load_parameters_scalar(real_nr)
+        df_transformed = df.filter(pl.col("type") == "transformed").drop("type")
+        transformed_dict = {}
+
+        for col in df_transformed.columns:
+            group_name, param_name = col.split(":")
+            transformed_value = df_transformed.select(pl.col(col)).to_series()[0]
+
+            # Build the nested dictionary
+            if group_name not in transformed_dict:
+                transformed_dict[group_name] = {}
+            transformed_dict[group_name][param_name] = transformed_value
+        # todo log handling when distribution requires it
```

Comment:
TODO: here is also the place that we need to handle template_file / output_file!